### PR TITLE
Bg_ingate OK

### PR DESCRIPTION
**Pull request by jpburnett**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

Time to take a tour along the swamp!

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c [hunk 1]
```diff
@@ -28,41 +38,357 @@ const ActorInit Bg_Ingate_InitVars = {
...
+            }
+        }
+    } else if (DynaPolyActor_IsInRidingMovingState(&this->dyna) == 0) {
+        this->unk160 |= 4;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ingate/func_80953EA4.s")
+void func_809542A0(BgIngate* this, GlobalContext* globalCtx) {
+    if ((gSaveContext.eventInf[5] & 1) != 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c:246]:
```suggestion
    if (gSaveContext.eventInf[5] & 1) {
```

File: src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c [hunk 2]
```diff
@@ -28,41 +38,357 @@ const ActorInit Bg_Ingate_InitVars = {
...
+        globalCtx->nextEntranceIndex = 0xA820;
+        gSaveContext.eventInf[5] &= 0xFE;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c:248]:
```suggestion
        gSaveContext.eventInf[5] &= (u8)~1;
```

File: src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c [hunk 3]
```diff
@@ -28,41 +38,357 @@ const ActorInit Bg_Ingate_InitVars = {
...
+    } else {
+        globalCtx->nextEntranceIndex = 0xA810;
+    }
+    gSaveContext.nextCutsceneIndex = 0;
+    globalCtx->sceneLoadFlag = 0x14;
+    globalCtx->unk_1887F = 3;
+    gSaveContext.nextTransition = 3;
+    this->actionFunc = func_80953F8C;
+    gSaveContext.weekEventReg[90] &= 0xBF;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c:257]:
```suggestion
    gSaveContext.weekEventReg[90] &= (u8)~0x40;
```

File: src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c [hunk 4]
```diff
@@ -28,41 +38,357 @@ const ActorInit Bg_Ingate_InitVars = {
...
+    func_800FE498();
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ingate/func_80953F14.s")
+void func_80954340(BgIngate* this, GlobalContext* globalCtx) {
+    if (!DECR(this->unk16A)) {
+        if (this->unk164 != 0) {
+            func_800B7298(globalCtx, &this->dyna.actor, 6U);
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c:264]:
```suggestion
            func_800B7298(globalCtx, &this->dyna.actor, 6);
```

File: src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c [hunk 5]
```diff
@@ -28,41 +38,357 @@ const ActorInit Bg_Ingate_InitVars = {
...
+            this->unk164 = &globalCtx->setupPathList[this->unk164->unk1];
+            func_80953F14(this, globalCtx);
+            func_800FE484();
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ingate/func_80953F8C.s")
+void func_809543D4(BgIngate* this, GlobalContext* globalCtx) {
+    u8 talkState;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c:273]:
```suggestion
    u8 talkState = Message_GetState(&globalCtx->msgCtx);
```
Would also delete the `talkState = Message_GetState(&globalCtx->msgCtx);` below.

File: src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c [hunk 6]
```diff
@@ -28,41 +38,357 @@ const ActorInit Bg_Ingate_InitVars = {
...
+                    this->unk160 &= ~0x4;
+                    this->actionFunc = func_809541B8;
+                    func_800FE498();
+                    func_8019F208();
+                } else {
+                    if (this->unk164 != 0) {
+                        this->unk164 = &globalCtx->setupPathList[this->unk164->unk1];
+                    }
+                    func_80953F14(this, globalCtx);
+                    gSaveContext.weekEventReg[0x5A] &= 0xBF;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c:294]:
```suggestion
                    gSaveContext.weekEventReg[0x5A] &= (u8)~0x40;
```

File: src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c [hunk 7]
```diff
@@ -28,41 +38,357 @@ const ActorInit Bg_Ingate_InitVars = {
...
+                    func_8019F230();
+                }
+                func_801477B4(globalCtx);
+                break;
+            case 0x9E6:
+                if (globalCtx->msgCtx.choiceIndex == 0) {
+                    func_80953EA4(this, globalCtx);
+                    gSaveContext.weekEventReg[0x5A] &= 0xBF;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c:302]:
```suggestion
                    gSaveContext.weekEventReg[0x5A] &= (u8)~0x40;
```

File: src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c [hunk 8]
```diff
@@ -28,41 +38,357 @@ const ActorInit Bg_Ingate_InitVars = {
...
+                func_801477B4(globalCtx);
+                break;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ingate/func_809541B8.s")
+void BgIngate_Init(Actor* thisx, GlobalContext* globalCtx2) {
+    BgIngate* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c:320]:
```suggestion
    GlobalContext* globalCtx = globalCtx2;
    BgIngate* this = THIS;
```
Does this still match?

File: src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c [hunk 9]
```diff
@@ -28,41 +38,357 @@ const ActorInit Bg_Ingate_InitVars = {
...
+        DynaPolyActor_Init(&this->dyna, 3);
+        DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &gSichitaiBoatCol);
+        this->unk160 = 0;
+        this->unk160 |= 0x8;
+        this->unk160 |= 0x10;
+        Actor_SetScale(&this->dyna.actor, 1.0f);
+        this->unk164 = func_8013BB34(globalCtx, BGINGATE_GET_FF(&this->dyna.actor), 0);
+        this->dyna.actor.room = -1;
+        if (gSaveContext.weekEventReg[0x14] & 2) {
+            gSaveContext.weekEventReg[0x5A] &= 0xBF;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c:336]:
```suggestion
            gSaveContext.weekEventReg[0x5A] &= (u8)~0x40;
```

File: src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c [hunk 10]
```diff
@@ -28,41 +38,357 @@ const ActorInit Bg_Ingate_InitVars = {
...
+        }
+        if ((!(gSaveContext.eventInf[3] & 0x20)) && (gSaveContext.weekEventReg[0x5A] & 0x40)) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c:338]:
```suggestion
        if (!(gSaveContext.eventInf[3] & 0x20) && (gSaveContext.weekEventReg[0x5A] & 0x40)) {
```

File: src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.h [hunk 11]
```diff
@@ -7,11 +7,23 @@ struct BgIngate;
...
+
 typedef struct BgIngate {
-    /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0x18];
-    /* 0x015C */ BgIngateActionFunc actionFunc;
-    /* 0x0160 */ char unk_160[0x30];
+    /* 0x000 */ DynaPolyActor dyna;
+    /* 0x15C */ BgIngateActionFunc actionFunc;
+    /* 0x160 */ u16 unk160;
+    /* 0x164 */ Path *unk164;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.h:16]:
```suggestion
    /* 0x164 */ Path* unk164;
```

**Review by hensldm — CHANGES_REQUESTED** [review]
Mostly just a bunch of little things, but need to address the player stateflags.

File: src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c (hunk 9 above)
Comment (hensldm) [reply src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c:334]: Currently don't see this as changed yet.

File: src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c (hunk 4 above)
Comment (hensldm) [reply src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c:263]: Currently don't see this as changed yet.

**Review by hensldm — APPROVED** [review]
There were two things that got marked as resolved that didn't seem to be changed, but otherwise looks good.

File: src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c (hunk 9 above)
Comment (jpburnett) [reply src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c:334]: Hmm, I guess my find and replace didn't catch everything. should be fixed now

**Review by AngheloAlf — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**